### PR TITLE
issues-686 repr in index.py to return more info

### PR DESCRIPTION
**Pull request by shreeju**
modified repr in index.py to return dense shape, charges and flow inf…ormation

### CODE REVIEW

**Comment by shreeju** [comment]
@googlebot I signed it!

**Comment by Thenerdstation** [comment]
First pass makes me think we'll want a slightly more formatting, but I'll leave that up to martin.

**Comment by mganahl** [comment]
I agree with Chase, the formatting right now is not great. In general, please add a meaningful descriptor to each output, instead of just the value, something like 
```
dimension: 10
```
instead of  just
```
10
```

File: tensornetwork/block_sparse/index.py [hunk 1]
```diff
@@ -44,7 +44,10 @@ def __len__(self) -> int:
     return self.dim
 
   def __repr__(self) -> str:
-    return str(self.dim)
+    dense_shape = str(self.dim)
```
Comment (mganahl) [inline tensornetwork/block_sparse/index.py:47]: add a heading, something like "Index:"
and then with some inset add the actual data
als add descriptors: "dimension", "charge", "flows"

Comment (shreeju) [reply tensornetwork/block_sparse/index.py]: Thank you. I have added it now.

Comment (mganahl) [reply tensornetwork/block_sparse/index.py]: Thanks @shreeju! I submitted an PR to your PR with some minor modifications. Should be good once you pull it in

**Comment by shreeju** [comment]
@googlebot I consent

**Comment by mganahl** [comment]
@googlebot I consent

**Pull request merged**